### PR TITLE
fix some misc in pilot

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
1. remove redundant checks

1. use `k8s.io/apimachinery/pkg/api/errors` to check kube-apiserver not found error

### CODE REVIEW

File: pilot/pkg/bootstrap/server.go [hunk 1]
```diff
@@ -209,11 +210,7 @@ func NewServer(args PilotArgs) (*Server, error) {
 		args.Namespace = os.Getenv("POD_NAMESPACE")
 	}
 	if args.Config.ClusterRegistriesNamespace == "" {
-		if args.Namespace != "" {
```
Comment (hzxuzhonghu) [inline pilot/pkg/bootstrap/server.go]: args.Namespace is not nil with above.

File: pilot/pkg/bootstrap/server.go [hunk 2]
```diff
@@ -311,9 +308,7 @@ func (s *Server) initClusterRegistries(args *PilotArgs) (err error) {
 			return err
 		}
 	}
-	if s.clusterStore != nil {
```
Comment (hzxuzhonghu) [inline pilot/pkg/bootstrap/server.go]: ` s.clusterStore != nil` is absolutely not nil.

File: pilot/pkg/bootstrap/server.go [hunk 3]
```diff
@@ -350,7 +345,7 @@ func GetMeshConfig(kube kubernetes.Interface, namespace, name string) (*v1.Confi
 
 	config, err := kube.CoreV1().ConfigMaps(namespace).Get(name, meta_v1.GetOptions{})
 	if err != nil {
-		if strings.Contains(err.Error(), fmt.Sprintf("\"%s\" not found", name)) {
+		if errors.IsNotFound(err) {
```
Comment (hzxuzhonghu) [inline pilot/pkg/bootstrap/server.go]: use standard kube apimachinery way to check not found error.

File: pilot/pkg/bootstrap/server.go [hunk 4]
```diff
@@ -379,17 +374,16 @@ func (s *Server) initMesh(args *PilotArgs) error {
 		return nil
 	}
 	var mesh *meshconfig.MeshConfig
+	var err error
+
 	if args.Mesh.ConfigFile != "" {
-		fileMesh, err := cmd.ReadMeshConfig(args.Mesh.ConfigFile)
```
Comment (hzxuzhonghu) [inline pilot/pkg/bootstrap/server.go]: reduce one else beanch

**Comment by rshriram** [comment]
Release 1.0 branch.

File: pilot/pkg/bootstrap/server.go (hunk 1 above)
Comment (costinm) [reply pilot/pkg/bootstrap/server.go]: Why ? May run in non-k8s env

Comment (hzxuzhonghu) [reply pilot/pkg/bootstrap/server.go]: Oh, I missed that.

**Comment by hzxuzhonghu** [comment]
>However this is the kind of change that would be better of against master ( which should be again open for business soon ), not P0/critical for 1.0.

ok

**Comment by hzxuzhonghu** [comment]
@ZackButcher https://github.com/istio/istio/pull/7000#pullrequestreview-137670572 suggested  open this PR against the master. So I am confused.

**Comment by hzxuzhonghu** [comment]
ping @ZackButcher  @rshriram ptal

**Pull request merged**